pkg/publisher/publisher.go

```diff
@@ -73,6 +76,30 @@ func FindIface(ip net.IP) (iface net.Interface, err error) {
 	return iface, fmt.Errorf("Couldn't find interface with IP address %s", ip)
 }
 
+// If an interface changes after we've started listening on it (say because it
+// got bridged), we need to be restarted to listen on the new interface that
+// contains the specified IP.
+func IfaceCheck(ip net.IP, iface net.Interface) {
+	for {
+		i, err := FindIface(ip)
+		if err != nil || i.Name != iface.Name {
```

Comment:
Let's say we fail to find the interface since the IP is being moved and does not sit anywhere at the moment. What would happen with mdns on the next start?

Reply:
I confirmed that if the IP is not found mdns-publisher just exits immediately, so it would crash loop in that case until the IP gets configured.